source/textUtils.py

```diff
@@ -247,3 +305,126 @@ def isLowSurrogate(ch: str) -> bool:
 # used to replace an unknown, unrecognized, or unrepresentable character.
 # https://en.wikipedia.org/wiki/Specials_(Unicode_block)
 REPLACEMENT_CHAR = u"\uFFFD"
+
+
+class UTF8OffsetConverter(OffsetConverter):
+	R"""
+	Object that holds a string in both its decoded and its UTF-8 encoded form.
+	The object allows for easy conversion between offsets in str type strings,
+	and offsets in UTF-8 encoded strings.
+
+	A single character in UTF-8 encoding might take 1, 2, or 4 bytes.
+	Examples of applications using UTF-8 encoding are all Scintilla-based text editors,
+	including Notepad++ and AkelPad.

```

Comment:
AkelPad is not Scintilla based and uses its own edit control. This comment should probably be clarified.